cupy/cuda/stream.pyx

```diff
@@ -314,6 +349,8 @@ class Stream(BaseStream):
 
     Attributes:
         ~Stream.ptr (intptr_t): Raw stream handle.
+        ~Stream.dev (int): The ID of the device that the stream was created on.
+            The value ``-1`` is used for the singleton stream objects.
```

Comment:
If we create a singleton object per device, the `dev` attribute could be more meaningful, saving us a few lookups. However, it'd be a breaking change which I am not sure if acceptable or not.

Reply:
I guess reducing `cudaGetDevice` lookup is difficult as CuPy may be used in conjunction with other libraries. For example CuPy code may be placed under `with torch.cuda.device(1):`.